### ISSUE TITLE
Variable-size pages?

### ISSUE THREAD

**Issue by ccleve** [orioledb/orioledb#10]
https://github.com/orioledb/orioledb/blob/main/doc/arch.md#data-compression says that storage pages have variable length. 

How did you pull that off? So far as I can tell, pg uses BLCKSZ pretty much everywhere. I looked through the code and I'm not seeing how you did it.

Did you somehow replace md.c in the storage manager?

I'm not seeing any commits here (https://github.com/orioledb/postgres/commits/patches14) which implement variable-length pages, but perhaps I missed it.

ORIOLEDB_BLCKSZ and ORIOLEDB_COMP_BLCKSZ are fixed.

**Comment by ccleve** [comment]
So does that mean you're still living inside standard 8k pages, but that you're allocating virtual extents within that structure?

**Comment by akorotkov** [comment]
Yes. In-memory pages are uncompressed pages and fixed 8k size. On-disk pages might be compressed and then they have variable sizes.
